src/Interpreters/Cache/FileCache.cpp

```diff
@@ -521,7 +521,7 @@ KeyMetadata::iterator FileCache::addFileSegment(
         result_state = state;
     }
 
-    auto file_segment = std::make_shared<FileSegment>(key, offset, size, result_state, settings, this, locked_key.getKeyMetadata());
+    auto file_segment = std::make_shared<FileSegment>(key, offset, size, result_state, settings, background_download_threads > 0, this, locked_key.getKeyMetadata());
```

Comment:
`0` usually means unlimited threads, but here it means we don't need to download a segment. Probably requires a comment somewhere.

Reply:
Added